subprojects/dependency-management/src/main/java/org/gradle/internal/resource/transfer/DefaultCacheAwareExternalResourceAccessor.java

```diff
@@ -155,16 +155,16 @@ public ExternalResourceMetaData create() {
         });
     }
 
-    private HashValue getResourceSha1(ExternalResourceName location, boolean revalidate) {
+    private HashCode getResourceSha1(ExternalResourceName location, boolean revalidate) {
         try {
             ExternalResourceName sha1Location = location.append(".sha1");
             ExternalResource resource = delegate.resource(sha1Location, revalidate);
-            ExternalResourceReadResult<HashValue> result = resource.withContentIfPresent(new Transformer<HashValue, InputStream>() {
+            ExternalResourceReadResult<HashCode> result = resource.withContentIfPresent(new Transformer<HashCode, InputStream>() {
                 @Override
-                public HashValue transform(InputStream inputStream) {
+                public HashCode transform(InputStream inputStream) {
                     try {
                         String sha = IOUtils.toString(inputStream, "us-ascii");
-                        return HashValue.parse(sha);
+                        return HashUtil.parse(sha, 40);
```

Comment:
Why the padding to 40? I didn't see that in the original code?

Reply:
It appears that we support SHA-1 hash codes that are shorter than 40 characters. We used to compare the values as `BigInteger`, which we don't do anymore. That's why we need the padding.